packages/server-side-render/src/server-side-render.js

```diff
@@ -77,39 +77,23 @@ export class ServerSideRender extends Component {
 						errorMsg: error.message,
 					} } );
 				}
+			} )
+			.finally( () => {
+				fetchComplete( this );
 			} );
 		return fetchRequest;
 	}
 
 	render() {
 		const response = this.state.response;
-		const { className } = this.props;
+		const { className, emptyResponsePlaceholder, errorResponsePlaceholder, loadingResponsePlaceholder } = this.props;
+
 		if ( response === '' ) {
-			return (
-				<Placeholder
-					className={ className }
-				>
-					{ __( 'Block rendered as empty.' ) }
-				</Placeholder>
-			);
+			return emptyResponsePlaceholder( this.props, response );
```

Comment:
I don't think this should be a callback.  Just make it a render prop.  So consumers would provide the component they want rendered here and it would be implemented via:

```jsx
<emptyResponsePlaceholder response={ response } { ...this.props } />
```

or more conventionally via (which is more similar to what you have):

```js
onEmptyResponsePlaceholder( { ...this.props, response } );
```

Reply:
Take a look now - I just made this change. I renamed the props because the component names required a capital.